### PR TITLE
[JENKINS-25519] If unable to read exitStatus file then wait briefly and retry one more time to allow write to complete

### PR DESCRIPTION
**Pull request by svanoort**
Proposed solution to error like this, encountered when statusCode file exists but is empty, potentially when created but not written to yet, or not written fully:

> java.lang.NumberFormatException: For input string: ""
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:592)
	at java.lang.Integer.parseInt(Integer.java:615)
	at org.jenkinsci.plugins.durabletask.FileMonitoringTask$FileMonitoringController.exitStatus(FileMonitoringTask.java:168)
> Caused: java.io.IOException: corrupted content in $SOMEPLACE 
> at org.jenkinsci.plugins.durabletask.FileMonitoringTask$FileMonitoringController.exitStatus(FileMonitoringTask.java:170)
	at org.jenkinsci.plugins.durabletask.BourneShellScript$ShellController.exitStatus(BourneShellScript.java:211)

Variant of [JENKINS-25519](https://issues.jenkins-ci.org/browse/JENKINS-25519)

This may not be the *best* solution to the problem in my opinion but should generally solve it.

Other options:
* Return null until the file parses cleanly (normal timeout will apply once the launching script dies and log file no longer is touched).
* Write and look for a delimiter on the exitStatus file to indicate that the file is fully written (avoids theoretical issues with only '1' being written when status code might actually be '102' or something).

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java [hunk 1]
```diff
@@ -163,11 +163,17 @@ protected FileMonitoringController(FilePath ws) throws IOException, InterruptedE
         // TODO would be more efficient to allow API to consolidate writeLog with exitStatus (save an RPC call)
         @Override public Integer exitStatus(FilePath workspace, Launcher launcher, TaskListener listener) throws IOException, InterruptedException {
             FilePath status = getResultFile(workspace);
-            if (status.exists()) {
+            if (status.exists() && status.length() > 0) {
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java]: Well this alone would fix the issue, right?

Comment (svanoort) [reply src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java]: Theoretically, yes, if there's no bogus whitespace or other stuff in there.

**Comment by jglick** [comment]
Ah, maybe I was thinking of #37—similar problem (at least on the face of it), different file.

If the problem is indeed a simple race condition, tricks like this are not the best fix. Rather, the wrapper script should be changed from

```sh
…; echo $? > …/exit
```

to

```
…; echo $? > …/exit.tmp; mv …/exit.tmp …/exit
```

which ought to be atomic.

**Comment by svanoort** [comment]
@jglick Has requested a test-comment, please ignore this.  @cloudbees/team-arc

**Comment by svanoort** [comment]
Closing in favor of #66

**Pull request closed without merging**